provider/dyn.go

```diff
@@ -387,6 +349,72 @@ func (d *dynProviderState) fetchZoneSerial(client *dynect.Client, zone string) (
 	return resp.Data.Serial, nil
 }
 
+//Use SOAP to fetch all records with a single call
+func (d *dynProviderState) fetchAllRecordsInZone(zone string) (*dynectsoap.GetAllRecordsResponseType, error) {
+	var err error
+	client := dynectsoap.NewClient("https://api2.dynect.net/SOAP/")
+	service := dynectsoap.NewDynect(client)
+
+	sessionRequest := dynectsoap.SessionLoginRequestType{
+		Customer_name:  d.CustomerName,
+		User_name:      d.Username,
+		Password:       d.Password,
+		Fault_incompat: 0,
+	}
+	resp := dynectsoap.SessionLoginResponseType{}
+	err = apiRetryLoop(func() error {
+		return service.Do(&sessionRequest, &resp)
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	token := resp.Data.Token
```

Comment:
If you pass dynect.Client to this method, I think you can reuse it's token and don't do a another login call. Well, assuming a rest token is valid for the soap endpoint.
If so, the logout call can also be saved.